runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorker.java

```diff
@@ -701,6 +414,71 @@ public void run() {
     LOG.debug("maxWorkItemCommitBytes: {}", maxWorkItemCommitBytes);
   }
 
+  /** Returns whether an exception was caused by a {@link OutOfMemoryError}. */
+  private static boolean isOutOfMemoryError(Throwable t) {
+    while (t != null) {
+      if (t instanceof OutOfMemoryError) {
+        return true;
+      }
+      t = t.getCause();
+    }
+    return false;
+  }
+
+  private static MapTask parseMapTask(String input) throws IOException {
+    return Transport.getJsonFactory().fromString(input, MapTask.class);
+  }
+
+  public static void main(String[] args) throws Exception {
```

Comment:
@kennknowles out of curiosity, do you happen to know where this is referenced?  I was looking for it in a gradle file, but couldn't find it and not sure if its even referenced in those.

Thanks!

Reply:
It is referenced in dataflow internal launching of the Dataflow v1 Java worker (not in github repo).
With beam portability, the runner specifics like this are behind the FnApi but for the v1 harness the dataflow details leaked some.  They were initially separate repositories but we kept having jar compatibility issues so we moved it to the external beam repository and single jar.